### PR TITLE
Fix mipmapping regression

### LINKED ISSUE #96: [bug] Regression in mipmapping calcs
**Describe the bug**
Regression has occurred (likely when computing triangle area) after the MDLParser integration that affects mipmapping

**To Reproduce**
Steps to reproduce the behaviour:
1. Use an existing ray tracer
2. Set the ray cone params when calling traverse to enable texture filtering

**Screenshots**
image

### PR DESCRIPTION
**Pull request by yogwoggf**
Fixes #96 

**PR Type (tick all that are applicable)**
- [x] Bug Fix
- [ ] New Feature (doesn't break Module <-> Lua compatibility)
- [ ] New Feature (breaks Module <-> Lua compatibility)
- [ ] Documentation Update Required

**Tested Targets (only applicable for changes to the binary module, delete otherwise)**
- [x] windows-x64-release
- [ ] windows-x86-release

**Checklist**
- [x] I have read and understand the contributing guidelines
- [x] I have tested all aspects of this PR (not required)

**Description**
Properly passes the UVs of any world/model triangle into the `TriangleBackfaceCull` class so that the data is intact once it calls `ComputeNormalAndLoD()`

### CODE REVIEW

File: source/objects/Model.cpp [hunk 1]
```diff
@@ -94,7 +101,7 @@ Mesh::Mesh(
 								tri.tangents[j] = glm::normalize(glm::vec3(tri.e1[0], tri.e1[1], tri.e1[2]));
 							}
 
-							tri.uvs[j] = glm::vec2(verts[j]->texCoord.x, verts[j]->texCoord.y);
+							tri.uvs[j] = uvs[j];
```
Comment (Derpius) [inline source/objects/Model.cpp:104]: this shouldn't be needed now that it's passed to the constructor right?

File: source/objects/Primitives.h [hunk 2]
```diff
@@ -75,9 +77,14 @@ struct TriangleBackfaceCull
 		const bvh::Vector3<Scalar> p1,
 		const bvh::Vector3<Scalar> p2,
 		const int16_t material,
+		const glm::vec2 uvs[3],
 		const bool oneSided = false
-	) : p0(p0), e1(p0 - p1), e2(p2 - p0), material(material), oneSided(oneSided)
+	) : p0(p0), e1(p0 - p1), e2(p2 - p0), material(material), uvs(), oneSided(oneSided)
```
Comment (Derpius) [inline source/objects/Primitives.h:82]: is this empty initialisation doing anything?

File: source/objects/AccelStruct.cpp [hunk 3]
```diff
@@ -411,7 +412,6 @@ World::World(GarrysMod::Lua::ILuaBase* LUA, const std::string& mapName)
 
 		memcpy(tri.normals, normals + vi0, sizeof(glm::vec3) * 3);
 		memcpy(tri.tangents, tangents + vi0, sizeof(glm::vec3) * 3);
-		memcpy(tri.uvs, uvs + vi0, sizeof(glm::vec2) * 3);
 		memcpy(tri.alphas, alphas + vi0, sizeof(float) * 3);
```
Comment (Derpius) [inline source/objects/AccelStruct.cpp:415]: Would be ideal to pass the other data in via the constructor too, but not gonna make you do that

**Review by Derpius — COMMENTED** [review]
Couple of small comments

**Comment by Derpius** [comment]
No idea why the build is failing now due to vcpkg (the repository has never used it)

File: source/objects/Model.cpp (hunk 1 above)
Comment (yogwoggf) [reply source/objects/Model.cpp:104]: Yep, my mistake

File: source/objects/Primitives.h (hunk 2 above)
Comment (yogwoggf) [reply source/objects/Primitives.h:82]: No

**Pull request merged**